File: src/main/java/edu/kit/iti/formal/stvs/model/table/SpecificationColumn.java
package edu.kit.iti.formal.stvs.model.table;

import java.util.List;

import javafx.beans.property.SimpleStringProperty;
import javafx.beans.property.StringProperty;

/**
 * A column in a specification (see {@link SpecificationTable}). The generic type parameter C is the
 * type of the cells.
 *
 * @author Benjamin Alt
 */
public class SpecificationColumn<C> implements Commentable {

  private List<C> cells;
  private StringProperty comment;

  /**
   * Create a new SpecificationColumn fro a list of cells.
   *
   * @param cells The cells for this column.
   */
  public SpecificationColumn(List<C> cells) {
    this.cells = cells;
    this.comment = new SimpleStringProperty("");
  }

  public List<C> getCells() {
    return cells;
  }

  @Override
  public void setComment(String comment) {
    this.comment.set(comment);
  }

  @Override
  public String getComment() {
    return this.comment.get();
  }

  @Override
  public StringProperty commentProperty() {
    return this.comment;
  }

  @Override
  public int hashCode() {
    int result = getCells() != null ? getCells().hashCode() : 0;
    result = 31 * result + (getComment() != null ? getComment().hashCode() : 0);
    return result;
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj) {
      return true;
    }
    if (obj == null || getClass() != obj.getClass()) {
      return false;
    }

    SpecificationColumn<?> that = (SpecificationColumn<?>) obj;

    if ((getCells() != null)
        ? !getCells().equals(that.getCells()) : (that.getCells() != null)) {
      return false;
    }
    return (getComment() != null
        ? getComment().equals(that.getComment()) : that.getComment() == null);
  }

  @Override
  public String toString() {
    return "SpecificationColumn(cells: " + cells + ", comment: " + comment.get() + ")";
  }
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/model/table/SpecificationColumn.kt
package edu.kit.iti.formal.stvs.model.table

import javafx.beans.property.SimpleStringProperty

/**
 * A column in a specification (see [SpecificationTable]). The generic type parameter C is the
 * type of the cells.
 *
 * @author Benjamin Alt
 */
data class SpecificationColumn<C>(val cells: List<C>) : Commentable {
    override val commentProperty = SimpleStringProperty("")
    override fun toString(): String {
        return "SpecificationColumn(cells: $cells, comment: $comment)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as SpecificationColumn<*>

        if (cells != other.cells) return false
        if (comment != other.comment) return false

        return true
    }

    override fun hashCode(): Int {
        var result = cells.hashCode()
        result = 31 * result + comment.hashCode()
        return result
    }


}
